uaclient/files.py

```diff
@@ -0,0 +1,296 @@
+        content = content.replace('"', "'")
+        content = util.redact_sensitive_logs(content)
+        content = content.replace("'", '"')
+        super().write(content)
+
+    def filter_tokens(self, schema, data):
+        keys_to_del = []
+        for k, v in data.items():
+            if k not in schema:
+                keys_to_del.append(k)
+                LOG.debug(
+                    "{key} not present in {schema}".format(
+                        key=k, schema=schema
+                    )
+                )
+            else:
+                if isinstance(v, dict):
+                    self.filter_tokens(schema[k], v)
+                elif isinstance(v, list):
+                    for i_v in v:
+                        if isinstance(i_v, dict):
+                            self.filter_tokens(schema[k][0], i_v)
+        for k in keys_to_del:
+            LOG.debug("Deleting {k} key from the data".format(k=k))
+            if isinstance(data, Dict):
+                del data[k]
+            else:
+                if k in data:
+                    data.remove(k)
+
+
+class MachineTokenFile:
+    _machine_token = None
+    _entitlements = None
+    _contract_expiry_datetime = None
+
+    def __init__(
+        self, root_directory: Optional[str] = None, root_mode: bool = True
+    ):
+        self._rd = None
+        self._fn = None
+        if root_mode:  # private file
+            if not root_directory:
+                root_directory = defaults.DEFAULT_DATA_DIR + "/private"
+            file_name = defaults.PRIVATE_MACHINE_TOKEN_FILE
+            self._rd = root_directory
+            self._fn = file_name
+            self.file_handler = UAFile(root_directory, self._fn)
+        else:  # public file
+            self.file_handler = PublicMachineTokenFile(root_directory)
+
+    def write(self, content):
+        """Update the machine_token file for both pub/priv files"""
+        self.file_handler.write(content)
+        if self.file_handler.is_private_file:  # write to public file as well
+            pub_file = PublicMachineTokenFile(self._rd)
+            pub_file.write(content)
+        self._machine_token = None
+        self._entitlements = None
+
+    def delete(self):
+        """Delete both pub and priv files"""
+        self.file_handler.delete()
+        # Delete the other file as well
+        if self.file_handler.is_private_file:
+            PublicMachineTokenFile(self._rd).delete()
+        else:
+            UAFile(self._rd, self._fn).delete()
+        self._machine_token = None
+        self._entitlements = None
+        self.file_handler = None
+
+    @property
+    def machine_token(self):
+        """Return the machine-token if cached in the machine token response."""
+        if not self._machine_token:
+            if not self.file_handler:
+                return None
+            content = self.file_handler.read()
+            try:
+                content = content.replace("'", '"')
```

Comment:
If we're writing single quotes to this file somewhere, then that is a separate bug that needs to be fixed. We should not try to support single quoted json because that's not json